ToolBox/GDAXDownloader/GDAXDownloaderProgram.cs

```diff
@@ -44,19 +45,25 @@ public static void GDAXDownloader(IList<string> tickers, string resolution, Date
             {
                 // Load settings from config.json
                 var dataDirectory = Config.Get("data-directory", "../../../Data");
+
+                // get symbol mapper for GDAX
+                var mapper = new SymbolPropertiesDatabaseSymbolMapper(Market.GDAX);
+
                 //todo: will download any exchange but always save as gdax
                 // Create an instance of the downloader
                 const string market = Market.GDAX;
                 var downloader = new GDAXDownloader();
-                var symbolMapper = new GDAXDownloaderSymbolMapper();
                 foreach (var ticker in tickers)
                 {
+                    // create symbols for lean and brokerage
+                    var leanSymbolObject = Symbol.Create(ticker, SecurityType.Crypto, market);
+                    var brokerageSymbolObject = Symbol.Create(mapper.GetBrokerageSymbol(leanSymbolObject), SecurityType.Crypto, market);
+
                     // Download the data
-                    var symbolObject = Symbol.Create(ticker, SecurityType.Crypto, market);
-                    var data = downloader.Get(symbolMapper.GetGDAXDownloadSymbol(symbolObject), castResolution, fromDate, toDate);
+                    var data = downloader.Get(brokerageSymbolObject, castResolution, fromDate, toDate);
```

Comment:
Generally, there isn't a `brokerageSymbolObject` but a brokerage ticker as shown in the `ISymbolMapper` interface and the 'services' it provides. Seems to me the `GDAXDownloader` should own the mapper internally, instead of relying on the caller doing the right thing, and we avoid the need of a `brokerageSymbolObject`

Reply:
make sense. updated